cmd/mailbox-controller/controller.go

```diff
@@ -255,76 +258,76 @@ func (ctl *mbCtl) sync(ctx context.Context, refany any) bool {
 func (ctl *mbCtl) ensureBinding(ctx context.Context, workspace *tenancyv1alpha1.Workspace) bool {
 	logger := klog.FromContext(ctx).WithValues("mbsName", workspace.Name)
 
-	// The script ensures that a mailbox workspace has the needed KubeStellar CRDs.
 	// The script must be idempotent.
-	shellScriptName := "kubectl-kubestellar-ensure-mbw"
+	shellScriptName := "kube-bind"
 
-	logger.V(2).Info("Executing shell script", "script", shellScriptName)
-	// Make a sandbox for the concurrent access of the kubeconfig from the mailbox-controller
-	cmdLine := "cp $KUBECONFIG $KUBECONFIG.copy && KUBECONFIG=$KUBECONFIG.copy " + shellScriptName + " " + workspace.Name
-	logger.V(2).Info(cmdLine)
-	cmd := exec.Command("/bin/sh", "-c", cmdLine)
-	stdout, err := cmd.StdoutPipe()
-	if err != nil {
-		logger.Error(err, "Unable to return a pipe for stdout")
-		return true
-	}
-	stderr, err := cmd.StderrPipe()
-	if err != nil {
-		logger.Error(err, "Unable to return a pipe for stderr")
-		return true
-	}
-	err = cmd.Start()
-	if err != nil {
-		logger.Error(err, "Unable to start executing the script")
-		return true
-	}
-	outBuf := bufio.NewReader(stdout)
-	errBuf := bufio.NewReader(stderr)
+	resourcesToBind := []string{"syncerconfigs", "edgesyncconfigs"}
+	for idx, resource := range resourcesToBind {
+		logger.V(2).Info("Ensuring binding", "script", shellScriptName, "resource", resource)
 
-	r, _ := regexp.Compile("🔒 (Created|Updated) secret.*namespace (.*)")
-	found, namespaceName := false, ""
-	for {
-		line, _, err := errBuf.ReadLine()
+		// Make sandboxes for the concurrent access of the kubeconfig from the mailbox-controller's multiple workers
+		// This is a compromise due to the situation that 1) we are using kcp and 2) we are running tests on bare process of this controller
+		workerIdx, _ := ctx.Value(ctxKey("worker")).(int)
+		settingKubeConfigForWorker := fmt.Sprintf("cp $KUBECONFIG $KUBECONFIG.copy%d && KUBECONFIG=$KUBECONFIG.copy%d", workerIdx, workerIdx)
+
+		cmdLine := strings.Join([]string{
+			settingKubeConfigForWorker,
+			shellScriptName,
+			workspace.Name,
+			resource,
+		}, " ")
+		if idx == 0 {
+			cmdLine = cmdLine + " --start-konnector true"
+		}
+		logger.V(2).Info(cmdLine)
+		cmd := exec.Command("/bin/sh", "-c", cmdLine)
+		stdout, err := cmd.StdoutPipe()
 		if err != nil {
-			logger.V(4).Info("End of stderr")
-			break
+			logger.Error(err, "Unable to return a pipe for stdout")
```

Comment:
"return" is not what is being attempted here.

Reply:
I copied this from the go doc of the `StderrPipe()` function which says 'returns a pipe'. May I know what should be the better message here?

Reply:
The general idea here is that the error message should be written from the perspective of the code using `StderrPipe` rather than the definition of `StderrPipe`. So while the latter talks about what `StderrPipe` "returns", the code using that is not in the business of "returning" the same thing. The error message should be about what problem the _using_ code is having.

Reply:
That's crystal clear. I see the point now. Changed in latest push.